qiskit/transpiler/target.py

```diff
@@ -1311,151 +1314,218 @@ def from_configuration(
             pulse_alignment = timing_constraints.pulse_alignment
             acquire_alignment = timing_constraints.acquire_alignment
 
-        qubit_properties = None
-        if backend_properties is not None:
-            # pylint: disable=cyclic-import
-            from qiskit.providers.backend_compat import qubit_props_list_from_props
-
-            qubit_properties = qubit_props_list_from_props(properties=backend_properties)
-
-        target = cls(
-            num_qubits=num_qubits,
-            dt=dt,
-            granularity=granularity,
-            min_length=min_length,
-            pulse_alignment=pulse_alignment,
-            acquire_alignment=acquire_alignment,
-            qubit_properties=qubit_properties,
-            concurrent_measurements=concurrent_measurements,
-        )
-        name_mapping = get_standard_gate_name_mapping()
+        required_operations = {"measure", "delay"}
+        all_instructions = set.union(set(basis_gates), required_operations)
+        unsupported_instructions = set()
+        faulty_qubits = set()
+        faulty_ops = set()
+        qiskit_inst_mapping = get_standard_gate_name_mapping()
         if custom_name_mapping is not None:
-            name_mapping.update(custom_name_mapping)
-
-        # While BackendProperties can also contain coupling information we
-        # rely solely on CouplingMap to determine connectivity. This is because
-        # in legacy transpiler usage (and implicitly in the BackendV1 data model)
-        # the coupling map is used to define connectivity constraints and
-        # the properties is only used for error rate and duration population.
-        # If coupling map is not specified we ignore the backend_properties
+            qiskit_inst_mapping.update(custom_name_mapping)
+
+        qiskit_control_flow_mapping = {
+            "if_else": IfElseOp,
+            "while_loop": WhileLoopOp,
+            "for_loop": ForLoopOp,
+            "switch_case": SwitchCaseOp,
+        }
+
+        inst_name_map = {}  # type: Dict[str, Instruction]
+        in_data_target = {
+            "num_qubits": num_qubits,
+            "granularity": granularity,
+            "min_length": min_length,
+            "pulse_alignment": pulse_alignment,
+            "acquire_alignment": acquire_alignment,
+            "dt": dt,
+            "concurrent_measurements": concurrent_measurements,
+        }
+
+        for name in all_instructions:
+            if name in qiskit_control_flow_mapping:
+                continue
+            if name in qiskit_inst_mapping:
+                inst_name_map[name] = qiskit_inst_mapping[name]
+            else:
+                warnings.warn(
```

Comment:
Yes, you are right keeping 'KeyError' is wise move here, because user might catch the error, add something in custom_name_mapping or basis_gates etc and run .from_configuration() all over again to get the target.
I thought I would change it to `TranspilerError`, if you do not like I would put `KeyError` back.